File: 剑指offer/32.1.从上到下打印二叉树.py
from typing import List


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def levelOrder(self, root: TreeNode) -> List[int]:
        ans = []

        if not root:
            return []

        queue = [root]
        while queue:
            cur = queue.pop(0)
            ans.append(cur.val)

            if cur.left:
                queue.append(cur.left)
            if cur.right:
                queue.append(cur.right)

        return ans

File: 剑指offer/32.1.从上到下打印二叉树.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x)
        : val(x)
        , left(NULL)
        , right(NULL)
    {
    }
};

class Solution {
public:
    vector<int> levelOrder(TreeNode* root)
    {
        vector<int> ans;
        queue<TreeNode*> q;
        if (!root) {
            return ans;
        }

        q.push(root);

        while (!q.empty()) {
            int sz = q.size();
            for (int i = 0; i < sz; i++) {
                auto cur = q.front();
                q.pop();

                ans.push_back(cur->val);

                if (cur->left) {
                    q.push(cur->left);
                }
                if (cur->right) {
                    q.push(cur->right);
                }
            }
        }
        return ans;
    }
};